src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -311,14 +313,32 @@ async def setup_pytest_for_target(
             f"--cov-config={coverage_config.path}",
             *cov_args,
         ]
+        pants_add_opts.extend(["--cov-report", "--cov-config"])  # --cov not there as multi-allowed.
+
+    # Get extra PYTEST_ADDOPTS env var from each possible sources.
```

Comment:
This was all starting to feel a bit like too much work / too brittle / etc. so I asked why are we doing things this way / is there another way? 

As to why, I traced history and can find nospecific justification for choosing env var vs directly using args:
+ origin of PYTEST_ADDOPTS: https://github.com/pantsbuild/pants/pull/8901
+ use of PYTEST_ADDOPTS to deal with junit xml: https://github.com/pantsbuild/pants/pull/9594

Here's why pytest has to say about PYTEST_ADDOPTS:
https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=pytest_addopts#envvar-PYTEST_ADDOPTS

The pre-pend bit made me want to know how multiple instance of a flag works then:
```
python3.9 -mvenv pytest.venv
pytest.venv/bin/pip -q install -U pip
pytest.venv/bin/pip install pytest==7.0.1
touch test.py
$ PYTEST_ADDOPTS="--junit-xml fred.xml" ./pytest.venv/bin/pytest --junit-xml george.xml --junitxml=betty.xml --collect-only -q test.py

------------------------------------------------------------------------------------------------------------------ generated xml file: /home/jsirois/betty.xml ------------------------------------------------------------------------------------------------------------------
no tests collected in 0.00s
```

So it looks like CLI trumps env var and right-most CLI arg trumps leftmost. Since we completely control the CLI args passed to pytest below, it seems we could just insert the args we require there. Our communication channel then becomes fully robust and looking for duplicates in PYTEST_ADDOPTS to warn is just a nicety at that point. In fact, reading about the `-o` option, it turns out its a very difficult nicety to unfailingly present! We only check env vars but we could be stomping a setting in config. We've been doing that all along it turns out! So maybe even the dup warnings checking is going to far, or at least we should only be checking for the `--junitxml` dup, since, FWICT, you cannot specify that option in config.

Reply:
I'm not sure I follow. There are so many options that the user can pass via `PYTEST_ADDOPTS` that it becomes difficult to check them all. That's why I simply put a warning. And what if the user really wants to override an option used/set by pants?

Reply:
At least, if there is weird behavior like the one I experienced, the user will have a little hint. In my case, I was completly in the dark.

Reply:
My main point is - can we use argv to pass the args we _must_ have never be over-ridden instead of using the current `PYTEST_ADDOPTS` mechanism for our own necessary options. That would fully solve robustness.

Then warning about collisions for those options we _must_ pass via the argv stomping user options is just a nicety. I was further pointing out its a nicety we do not extend to junitfamily today since that requires reading all potential config files where users might have set it. Further, doing so would be fragile - we'd be re-implementing the pytest config logic. As such, maybe warning should be jettisoned or at least restricted to just --junit-xml, which, FWICT, cannot be set via config file like junitfamily can.

Reply:
> At least, if there is weird behavior like the one I experienced, the user will have a little hint. In my case, I was completely in the dark.

And you'll still be completely in the dark if you use pytest config files in your repo to set junitfamily.

Reply:
Ok, I see... I'll wait for others to chime in before removing the warnings completely or restricting to only --junit-xml/--junitxml. Thank you for your feedback.

Reply:
Confirming that I looked at those older PRs and I see no strong reason they used the env var instead of the cli ags, other than the author's personal preference (or them being unware of the args possibility).